platform/security/security-oidc-realm/src/main/java/org/codice/ddf/security/oidc/realm/CustomOidcProfileCreator.java

```diff
@@ -47,85 +35,44 @@
-    profile.setIdTokenString(idToken.getParsedString());
 
-    try {
-      IDTokenClaimsSet claimsSet = new IDTokenClaimsSet(idToken.getJWTClaimsSet());
-      assertNotNull("claimsSet", claimsSet);
-      profile.setId(ProfileHelper.sanitizeIdentifier(profile, claimsSet.getSubject()));
-    } catch (java.text.ParseException | ParseException e) {
-      LOGGER.warn("Unable to extract ID token claim set.", e);
+    final AccessToken accessToken = credentials.getAccessToken();
+    if (accessToken != null && !accessToken.getValue().isEmpty()) {
+      profile.setAccessToken(accessToken);
     }
 
     final RefreshToken refreshToken = credentials.getRefreshToken();
     if (refreshToken != null && !refreshToken.getValue().isEmpty()) {
       profile.setRefreshToken(refreshToken);
-      LOGGER.debug("Refresh Token successful retrieved");
     }
 
-    try {
-      if (metadata.getUserInfoEndpointURI() != null && accessToken != null) {
-
-        final UserInfoRequest userInfoRequest =
-            new UserInfoRequest(metadata.getUserInfoEndpointURI(), (BearerAccessToken) accessToken);
-        final HTTPRequest userInfoHttpRequest = userInfoRequest.toHTTPRequest();
-        userInfoHttpRequest.setConnectTimeout(configuration.getConnectTimeout());
-        userInfoHttpRequest.setReadTimeout(configuration.getReadTimeout());
-        final HTTPResponse httpResponse = userInfoHttpRequest.send();
-        LOGGER.debug(
-            "Token response: status={}, content={}",
-            httpResponse.getStatusCode(),
-            httpResponse.getContent());
-
-        final UserInfoResponse userInfoResponse = UserInfoResponse.parse(httpResponse);
-        if (userInfoResponse instanceof UserInfoErrorResponse) {
-          LOGGER.error(
-              "Bad User Info response, error={}",
-              ((UserInfoErrorResponse) userInfoResponse).getErrorObject());
-        } else {
-          final JWTClaimsSet userInfoClaimsSet;
-          final UserInfoSuccessResponse userInfoSuccessResponse =
-              (UserInfoSuccessResponse) userInfoResponse;
-          if (userInfoSuccessResponse.getUserInfo() != null) {
-            userInfoClaimsSet = userInfoSuccessResponse.getUserInfo().toJWTClaimsSet();
-          } else {
-            userInfoClaimsSet = userInfoSuccessResponse.getUserInfoJWT().getJWTClaimsSet();
-          }
+    final JWT idToken = credentials.getIdToken();
+    profile.setIdTokenString(idToken.getParsedString());
 
-          getProfileDefinition().convertAndAdd(profile, userInfoClaimsSet.getClaims(), null);
-        }
-      }
+    try {
+      JWTClaimsSet claimsSet = idToken.getJWTClaimsSet();
+      assertNotNull("claimsSet", claimsSet);
+      profile.setId(ProfileHelper.sanitizeIdentifier(profile, claimsSet.getSubject()));
 
-      for (final Map.Entry<String, Object> entry :
-          idToken.getJWTClaimsSet().getClaims().entrySet()) {
-        if (!JwtClaims.SUBJECT.equals(entry.getKey())
```

Comment:
❓ Why was this condition removed?

Reply:
Thats a great question lol.

It pretty much checks to make sure duplicate information isn't added to the attributes, as well as making sure the subject string isn't added to the attributes (I think the reasoning for the subject thing is because it is stored elsewhere on the `OidcProfile`).

I mainly took that out for "make sure it works" purposes, but we can throw it back in! It may also be a good idea to figure out what we don't want in the `OidcProfile` attributes, since we may now want to filter more than just duplicates & the subject (for instance: an ID_Token).

Reply:
Yeah I think we should add this back then at least for now until we figure the rest out